# Displaying the result of Scraping the 50 pages of results 

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('ph_full.csv',delimiter=';',index_col=0)

In [3]:
df

pharmacie  \
0                Pharmacie  Oulmès   
1           Pharmacie  Oum Al Qora   
2          Pharmacie  Oum El Kheir   
3          Pharmacie  Oum ElBanine   
4           Pharmacie  Oum Errabii   
5          Pharmacie  Oum L'Ghaite   
6                Pharmacie  Oumnia   
7               Pharmacie  Oussama   
8               Pharmacie  Ouzineb   
9               Pharmacie  Oxygène   
10             Pharmacie  Panorama   
11          Pharmacie  Panoramique   
12              Pharmacie  Paranfa   
13           Pharmacie  parisienne   
14              Pharmacie  Pitchou   
15   Pharmacie  Place Ahmed Mekwar   
16         Pharmacie  Polyclinique   
17      Pharmacie  Pont de l'Oasis   
18            Pharmacie  populaire   
19                Pharmacie  Alabe   
20       Pharmacie  AlAhd El Jadid   
21              Pharmacie  Alahram   
22           Pharmacie  AlAkhawayn   
23               Pharmacie  AlAlia   
24               Pharmacie  AlAmal   
25              Pharmacie  Alamana   
26              Pharmacie  AlAmane   
27           Pharmacie  AlAndalous   
28        Pharmacie  AlAndaloussia   
29            Pharmacie  AlAntaria   
..                             ...   
949               Pharmacie  Bjani   
950         Pharmacie  Bloc Koudia   
951           Pharmacie  Bnou Zohr   
952    Pharmacie  Borj ElBernoussi   
953             Pharmacie  Bouabid   
954              Pharmacie  Bouafi   
955             Pharmacie  Boualam   
956          Pharmacie  Bouazzaoui   
957               Pharmacie  Bouih   
958            Pharmacie  Boukhari   
959        Pharmacie  du Croissant   
960      Pharmacie  du Dispensaire   
961            Pharmacie  du Gharb   
962      Pharmacie  du Golf d'Anfa   
963    Pharmacie  du Grand Maghreb   
964           Pharmacie  du Jasmin   
965             Pharmacie  du Jura   
966            Pharmacie  du Liban   
967           Pharmacie  du Louvre   
968            Pharmacie  du lycée   
969     Pharmacie  du Lycée Chawki   
970              Pharmacie  du Lys   
971           Pharmacie  du Maârif   
972             Pharmacie  du Mall   
973           Pharmacie  du Marché   
974   Pharmacie  du Marché de Gros   
975         Pharmacie  du Mechouar   
976            Pharmacie  du Métro   
977             Pharmacie  du midi   
978          Pharmacie  du Minaret   

                                                  lien  \
0    https://www.telecontact.ma/annonceur/pharmacie...   
1    https://www.telecontact.ma/annonceur/pharmacie...   
2    https://www.telecontact.ma/annonceur/pharmacie...   
3    https://www.telecontact.ma/annonceur/pharmacie...   
4    https://www.telecontact.ma/annonceur/pharmacie...   
5    https://www.telecontact.ma/annonceur/pharmacie...   
6    https://www.telecontact.ma/annonceur/pharmacie...   
7    https://www.telecontact.ma/annonceur/pharmacie...   
8    https://www.telecontact.ma/annonceur/pharmacie...   
9    https://www.telecontact.ma/annonceur/pharmacie...   
10   https://www.telecontact.ma/annonceur/pharmacie...   
11   https://www.telecontact.ma/annonceur/pharmacie...   
12   https://www.telecontact.ma/annonceur/pharmacie...   
13   https://www.telecontact.ma/annonceur/pharmacie...   
14   https://www.telecontact.ma/annonceur/pharmacie...   
15   https://www.telecontact.ma/annonceur/pharmacie...   
16   https://www.telecontact.ma/annonceur/pharmacie...   
17   https://www.telecontact.ma/annonceur/pharmacie...   
18   https://www.telecontact.ma/annonceur/pharmacie...   
19   https://www.telecontact.ma/annonceur/pharmacie...   
20   https://www.telecontact.ma/annonceur/pharmacie...   
21   https://www.telecontact.ma/annonceur/pharmacie...   
22   https://www.telecontact.ma/annonceur/pharmacie...   
23   https://www.telecontact.ma/annonceur/pharmacie...   
24   https://www.telecontact.ma/annonceur/pharmacie...   
25   https://www.telecontact.ma/annonceur/pharmacie...   
26   https://www.telecontact.ma/annonceur/pharmacie...   
27   https://www.telecontact.ma/annonceur/ph

In [4]:
df.describe()

pharmacie                                               lien  \
count                  979                                                979   
unique                 695                                                695   
top     Pharmacie  Annahda  https://www.telecontact.ma/annonceur/pharmacie...   
freq                     4                                                  4   

           access      coordonnee  
count         979             979  
unique        670             650  
top     NO ACCESS  NO COORDINATES  
freq           35              35

Telecontact says that it has 1230 results for all the pharmacies in Casablanca. But the page results are always random. 
Here we see that despite of the 979 different elements, w can notice that we have many duplicates

In [5]:
df['pharmacie'].duplicated()

0      False
1      False
2      False
3      False
4      False
5      False
6      False
7      False
8      False
9      False
10     False
11     False
12     False
13     False
14     False
15     False
16     False
17     False
18     False
19     False
20     False
21     False
22     False
23     False
24     False
25     False
26     False
27     False
28     False
29     False
       ...  
949    False
950    False
951    False
952    False
953    False
954    False
955    False
956    False
957    False
958    False
959     True
960     True
961     True
962     True
963     True
964     True
965     True
966    False
967    False
968    False
969    False
970    False
971    False
972    False
973    False
974    False
975    False
976    False
977    False
978    False
Name: pharmacie, Length: 979, dtype: bool

** Display all the duplicated pharmacies ** 

In [6]:
dup = pd.DataFrame()

In [7]:
dup['pharmacie'] = df['pharmacie']

In [8]:
dup['is_duplicate'] = df.duplicated()

In [13]:
redundance = dup[dup['is_duplicate'] == True]

In [14]:
redundance

pharmacie  is_duplicate
213                Pharmacie  Smaali          True
214                 Pharmacie  Smara          True
215                Pharmacie  Smirès          True
216               Pharmacie  Snoussi          True
217               Pharmacie  Socrate          True
218                Pharmacie  Sophia          True
268                 Pharmacie  Sevam          True
269             Pharmacie  Shahrazad          True
270     Pharmacie  Sidi Abderrahmane          True
271         Pharmacie  Sidi Elkhadir          True
272          Pharmacie  Sidi Maârouf          True
273           Pharmacie  Sidi Moumen          True
274           Pharmacie  Sidi Othman          True
275            Pharmacie  Sidi Soufi          True
276             Pharmacie  Sifeddine          True
277                 Pharmacie  Siham          True
278            Pharmacie  Sijilmassa          True
299             Pharmacie  Sifeddine          True
300                 Pharmacie  Siham          True
301            Pharmacie  Sijilmassa          True
302               Pharmacie  Sindbad          True
303                 Pharmacie  Skiba          True
304                Pharmacie  Slaoui          True
305                Pharmacie  Smaali          True
306                 Pharmacie  Smara          True
307                Pharmacie  Smirès          True
308               Pharmacie  Snoussi          True
309               Pharmacie  Socrate          True
310                Pharmacie  Sophia          True
311                 Pharmacie  Souad          True
..                               ...           ...
924                Pharmacie  Rihame          True
925                  Pharmacie  Rita          True
926            Pharmacie  Rive Bleue          True
927               Pharmacie  Rivièra          True
928     Pharmacie  Riyad Hay Hassani          True
929                  Pharmacie  Rizk          True
930                Pharmacie  Rochdi          True
931                Pharmacie  Rocroy          True
932              Pharmacie  Romandie          True
933  Pharmacie  Rosée de La Corniche          True
934               Pharmacie  Roudani          True
935      Pharmacie  Route d'Azemmour          True
936      Pharmacie  Route d'Eljadida          True
937        Pharmacie  Route de Rabat          True
938            Pharmacie  Ryad Sofia          True
939             Pharmacie  Bensabeur          True
940               Pharmacie  Bensaid          True
941              Pharmacie  Bensalah          True
942              Pharmacie  Bentaleb          True
943                Pharmacie  Benzit          True
944              Pharmacie  Beyrouth          True
945                 Pharmacie  Bilal          True
946          Pharmacie  Bin Lamdoune          True
959          Pharmacie  du Croissant          True
960        Pharmacie  du Dispensaire          True
961              Pharmacie  du Gharb          True
962        Pharmacie  du Golf d'Anfa          True
963      Pharmacie  du Grand Maghreb          True
964             Pharmacie  du Jasmin          True
965               Pharmacie  du Jura          True

[284 rows x 2 columns]